sbi/inference/snpe/snpe_c.py

```diff
@@ -150,12 +150,14 @@ def train(
             locals(), entries=("self", "__class__", "num_atoms", "use_combined_loss")
         )
 
-        # Set the proposal to the last proposal that was passed by the user. For atomic
-        # SNPE, it does not matter what the proposal is. For non-atomic SNPE, we only
-        # use the latest data that was passed, i.e. the one from the last proposal.
-        proposal = self._proposal_roundwise[-1]
-
-        if proposal is not None and proposal is not self._prior:
+        self._round = max(self._data_round_index)
```

Comment:
Note: this change is unrelated to this PR. The line:
```
if proposal is not None and proposal is not self._prior:
```
is also checked in `append_simulations` and results in a change in `self._data_round_index`. Thus, we just use `self._data_round_index` here instead of the if-case.